src/mngt/ocf_mngt_flush.c

```diff
@@ -917,6 +917,18 @@ struct ocf_mngt_cache_set_cleaning_context
 	void *priv;
 };
 
+static void _ocf_mngt_cleaning_deinit_complete(void *priv)
+{
+	struct ocf_mngt_cache_set_cleaning_context *context = priv;
+	ocf_cache_t cache = context->cache;
+
+	ocf_refcnt_freeze(&cache->cleaner.cleaning_ctx);
+	ocf_cleaning_deinitialize(cache);
+	ocf_refcnt_unfreeze(&cache->cleaner.cleaning_ctx);
```

Comment:
Don't we want to do unfreeze only after the new policy is initialized? Otherwise callbacks may be called before the new policy is initialized.